frontends/p4/simplifyParsers.cpp

```diff
@@ -113,6 +113,9 @@ class CollapseChains : public Transform {
         // has no other incoming edges.
         for (auto oe : *transitions) {
             auto node = oe.first;
+            if (!node->annotations->annotations.empty()) {
+                continue;
+            }
```

Comment:
Isn't this redundant with https://github.com/p4lang/p4c/pull/2452/files#diff-52d560b4ecf435649572dd0911559f23R130-R132?

Reply:
Consider a chain of states : s1 -> s2(@dontMerge) -> s3. 
s2 should not merge with s1 or s3. . Line 130 - 132 will avoid merging of s1 and s2. Line 116-118 will avoid merging of s2 and s3.

Reply:
Ah ha, I didn't notice the `node` vs. `next` difference. Thanks.